src/Guide/Types/Creds.hs

```diff
@@ -0,0 +1,92 @@
+{- |
+Module      :  Guide.Types.Creds
+Description :  Credentials for external providers (e.g.: GitHub authentication)
+Copyright   :  (c) Aaron Friel
+License     :  BSD-3
+
+Maintainer  :  Aaron Friel <mayreply@aaronfriel.com>
+Stability   :  unstable | experimental | provisional | stable | frozen
+Portability :  portable | non-portable (<reason>)
+
+-}
+
+module Guide.Types.Creds
+(
+  Creds,
+    credsProvider,
+    credsId,
+    credsExtra,
+  makeCreds,
+  canAddCreds,
+  PublicCreds,
+  credsToPublic,
+  publicCredsToCreds
+)
+where
+
+
+import Imports
+
+-- acid-state
+import Data.SafeCopy hiding (kind)
+import Data.SafeCopy.Migrate
+--
+import qualified Data.Map as Map
+
+data Creds = Creds {
+  -- | The external provider of the credentials (e.g.: github.com)
+  _credsProvider :: Text,
```

Comment:
Perhaps it makes sense to make it an ADT given that we only support a few providers?

Reply:
Some providers (e.g.: GitLab, Azure, Google GSuite etc.) might have multiple instances, disambiguated by the "name" parameter on the config.